OpenRA.Mods.Common/Traits/World/ResourceLayer.cs

```diff
@@ -95,6 +101,23 @@ public void WorldLoaded(World w, WorldRenderer wr)
 			}
 		}
 
+		void ITick.Tick(Actor self)
```

Comment:
I don't think you should tick on every `ResourceLayer` on every game on every mod when this is just a tiny TD campaign feature. Please only tick on the transform trait.

Reply:
The idea is to have it not only for the campaign but also for MP. The easiest solution would be to make a new subclass, ie `TDResouceLayer` and move the whole feature over theere,